packages/grafana-ui/src/components/GraphNG/GraphNG.tsx

```diff
@@ -212,16 +203,58 @@ export const GraphNG: React.FC<GraphNGProps> = ({
     );
   }
 
+  const legendItems = configBuilder
+    .getSeries()
+    .map<VizLegendItem | undefined>(s => {
+      const seriesConfig = s.props;
+      const fieldIndex = seriesConfig.dataFrameFieldIndex;
+      const axisPlacement = configBuilder.getAxisPlacement(s.props.scaleKey);
+
+      if (seriesConfig.hideInLegend || !fieldIndex) {
+        return undefined;
+      }
+
+      const field = data[fieldIndex.frameIndex]?.fields[fieldIndex.fieldIndex];
+
+      // I don't know what's going on... ;)
+      if (!field) {
+        return undefined;
```

Comment:
@dprokop I think I know why this happens, when the series count change you get one render here with new data, and old configBuilder.getSeries

Reply:
That's exactly what happens, I just can't think of any nicer way of handling this now. And this feels like something that might lead to confusing issues I believe. I mean, this pretty much stops anything wrong to happen as we are making sure the field exists, so undefined will not be referenced anywhere down the road.